test/e2e/framework/service_util.go

```diff
@@ -1450,3 +1461,28 @@ func GetServiceLoadBalancerCreationTimeout(cs clientset.Interface) time.Duration
 	}
 	return LoadBalancerCreateTimeoutDefault
 }
+
+func TestAffinityFromPod(execPod *v1.Pod, targetIpPort string) bool {
+	var hostname string
+	cmd := fmt.Sprintf(`wget -qO- http://%s/ -T 1`, targetIpPort)
+	affinityCount := 0
+	if pollErr := wait.PollImmediate(Poll, ServiceTestTimeout, func() (bool, error) {
+		stdout, err := RunHostCmd(execPod.Namespace, execPod.Name, cmd)
+		if err != nil || len(stdout) == 0 {
+			Logf("Failed to get response from %s. Retry until timeout", targetIpPort)
+			return false, nil
+		}
+		Logf("Got pod response: %s", stdout)
+		if hostname == stdout {
+			affinityCount++
+		} else {
+			affinityCount = 0
+			hostname = stdout
+		}
+		return affinityCount >= AffinityConfirmCount, nil
```

Comment:
Still thinking of early return when affinity does not hold. Probably we could return when X continuous requests hit different backends?

Reply:
Done. Now the function will return once it receives two different responses.